AjobK/Athena
# 282 delete posts

closes #284 

<br />

 **What is it supposed to do**  
Post owners and admins are able to delete posts from the front-end. These posts are soft deleted, meaning they're actually archived (implemented in another branch) and will still exist while not appearing anymore.

<br />

 **How can we test your branch**  
Create a post. While creating the post, the delete button should not be visible. After this, a delete button should appear at the bottom right.
![image](https://user-images.githubusercontent.com/14930277/132985718-c35b5806-a8d0-4841-9cbb-22fe2864622a.png)
The delete button should not appear on posts that aren't created by the user you're logged in on. It should also appear when you're an admin, but this can't be tested due to an external issue (see additional notes).

<br />

 **Additional notes**  
Backend:
In _postRoutes_ I replaced the _hasPermissions_ middleware at the _archivePost_ route with a new _hasPostPermissions_ middleware. This does the same as _hasPermissions_, but also checks if the user owns the post that is to be archived. As I'm not experienced with Express.js it might be worth looking at.

The pre-generated admin user appears to have a user role and not an admin role (this.props.store.profile.role returns 'user'). I wasn't able to test the delete functionality for admins due to this.

There is currently an issue in develop that crashes the application on startup, which should be fixed in the 1.1 hotfix branch.

<br />

**1. General checklist**

<br />

**2. If relevant, front-end checklist**

<br />

**3. If relevant, back-end checklist**

<br />

**4. If relevant, test these browsers**
